src/wallet/wallet.cpp

```diff
@@ -2775,6 +2775,12 @@ bool CWallet::CreateTransaction(const vector<CRecipient>& vecSend, CWalletTx& wt
 
                 // Check mempooltxinputlimit to avoid creating a transaction which the local mempool rejects
                 size_t limit = (size_t)GetArg("-mempooltxinputlimit", 0);
+                {
+                    LOCK(cs_main);
+                    if (NetworkUpgradeActive(chainActive.Height() + 1, Params().GetConsensus(), Consensus::UPGRADE_OVERWINTER)) {
```

Comment:
non-blocking: set variable `overwinterActive` ?

Reply:
I opted not to do this here, because `CreateTransaction` as currently inherited from upstream loops until it can calculate a "valid" fee, and depending on the number of UTXOs in the wallet, this could end up taking enough time that the upgrade boundary gets crossed mid-creation. Simpler to sample each round.